cmd/keymasterd/idp_oidc.go

```diff
@@ -129,16 +129,38 @@ func (state *RuntimeState) idpOpenIDCClientCanRedirect(client_id string, redirec
 		if client.ClientID != client_id {
 			continue
 		}
+		if len(client.AllowedRedirectDomains) < 1 && len(client.AllowedRedirectURLRE) < 1 {
+			return false, nil
+		}
+		matchedRE := false
 		for _, re := range client.AllowedRedirectURLRE {
 			matched, err := regexp.MatchString(re, redirect_url)
 			if err != nil {
 				return false, err
 			}
 			if matched {
-				return true, nil
+				matchedRE = true
+				break
+			}
+		}
+		// if no domains, the matchedRE ansmer is authoritative, no need to parse
+		if len(client.AllowedRedirectDomains) < 1 {
+			return matchedRE, nil
+		}
+		parsedURL, err := url.Parse(redirect_url)
+		if err != nil {
+			// TODO: Print ERR?
```

Comment:
Why not return an error rather than log it?

Reply:
Because the redirect_url is a parameter passed in by the client.  So most likely is an error due to client behavior.

Reply:
OK.